cache/manager.go

```diff
@@ -1099,6 +1099,8 @@ func calculateKeepBytes(totalSize int64, dstat disk.DiskStat, opt client.PruneIn
 		} else {
 			keepBytes = min(keepBytes, totalSize-excess)
 		}
+	} else if opt.MinFreeSpace != 0 && keepBytes == 0 {
+		keepBytes = totalSize
```

Comment:
might be good to have a comment here in keeping with the rest of the style of this function, to explain the logic of why this is here.